apps/admin/src/hooks/user-preferences.test.tsx

```diff
@@ -0,0 +1,288 @@
+const USER_UPDATE_API_URL = "/ghost/api/admin/users/:id/";
+
+// Test fixtures
+const fixtures = {
+    emptyPreferences: {},
+    existingPreferences: {
+        existing: "value",
+        shared: "old",
+    },
+    newPreferences: {
+        shared: "new",
+        additional: "data",
+    },
+    singlePreference: {
+        setting: "value",
+    },
+};
+
+// Type definitions
+type SetupUserPreferencesTest = (
+    userOverrides?: Partial<User>
+) => Promise<{ current: ReturnType<typeof useUserPreferencesQuery> }>;
+
+type SetupUpdatePreferencesTest = (initialUser?: Partial<User>) => Promise<{
+    query: { current: ReturnType<typeof useUserPreferencesQuery> };
+    mutation: { current: ReturnType<typeof useUpdateUserPreferences> };
+}>;
+
+const test = baseTest.extend<{
+    server: SetupServer;
+    queryClient: QueryClient;
+    wrapper: TestWrapperComponent;
+    setup: SetupUserPreferencesTest;
+    setupUpdate: SetupUpdatePreferencesTest;
+}>({
+    ...serverFixture,
+    ...queryClientFixtures,
+    setup: async (
+        {
+            server,
+            wrapper,
+        }: { server: SetupServer; wrapper: TestWrapperComponent },
+        provide: (setup: SetupUserPreferencesTest) => Promise<void>
+    ) => {
+        const setupUserPreferencesTest: SetupUserPreferencesTest = async (
+            userOverrides = {}
+        ) => {
+            server.use(
+                http.get(USERS_API_URL, () => {
+                    return HttpResponse.json({
+                        users: [
+                            {
+                                ...mockUser,
+                                ...userOverrides,
+                            },
+                        ],
+                    });
+                })
+            );
+
+            const { result } = renderHook(() => useUserPreferencesQuery(), {
+                wrapper,
+            });
+            await waitForQueryLoaded(result);
+            return result;
+        };
+        await provide(setupUserPreferencesTest);
+    },
+    setupUpdate: async (
+        {
+            server,
+            wrapper,
+        }: { server: SetupServer; wrapper: TestWrapperComponent },
+        provide: (setup: SetupUpdatePreferencesTest) => Promise<void>
+    ) => {
+        const setupUpdateTest: SetupUpdatePreferencesTest = async (
+            initialUser = {}
+        ) => {
+            server.use(
+                http.get(USERS_API_URL, () => {
+                    return HttpResponse.json({
+                        users: [
+                            {
+                                ...mockUser,
+                                ...initialUser,
+                            },
+                        ],
+                    });
+                }),
+                http.put<
+                    { id: string },
+                    UpdateUserRequestBody,
+                    UsersResponseType
+                >(USER_UPDATE_API_URL, async ({ request }) => {
```

Comment:
Maybe we can manually reformat this as the narrow print width makes the code hard to read.